pandas/core/strings.py

```diff
@@ -456,9 +456,10 @@ def str_replace(arr, pat, repl, n=-1, case=None, flags=0, regex=True):
     flags : int, default 0 (no flags)
         - re module flags, e.g. re.IGNORECASE
         - Cannot be set if `pat` is a compiled regex
-    regex : boolean, default True
+    regex : boolean, default None
         - If True, assumes the passed-in pattern is a regular expression.
         - If False, treats the pattern as a literal string
+        - If pat is single special character, default regex is False.
```

Comment:
This should mention something about a warning being issued when regex is not specified. Something like

    - If `pat` is a single character and `regex` is not specified, `pat` is interpreted as a
       string literal. If `pat` is also a regular expression symbol, a warning is issued that
       in the future `pat` will be interpreted as a regex, rather than a literal.